tests/integration/targets/hcloud_server_info/tasks/test.yml

```
@@ -1,127 +1,89 @@
 # Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 ---
-- name: setup ensure server is absent
-  hetzner.hcloud.hcloud_server:
-    name: "{{ hcloud_server_name }}"
-    state: absent
+- name: Gather hcloud_server_info
+  hetzner.hcloud.hcloud_server_info:
   register: result
-
-- name: create server
-  hetzner.hcloud.hcloud_server:
-    name: "{{ hcloud_server_name }}"
-    server_type: cx11
-    image: ubuntu-22.04
-    state: started
-    labels:
-      key: value
-  register: main_server
-- name: verify create server
-  assert:
+- name: Verify hcloud_server_info
+  ansible.builtin.assert:
     that:
-      - main_server is changed
-      - main_server.hcloud_server.name == "{{ hcloud_server_name }}"
-      - main_server.hcloud_server.server_type == "cx11"
-      - main_server.hcloud_server.status == "running"
-      - main_server.root_password != ""
+      - result.hcloud_server_info | list | count >= 1
```

Could be
```suggestion
      - result.hcloud_server_info | list | count >= 2
```

as we know that we have created 2 servers in `prepare.yml`, but really does not matter much.